pkg/blob/controllerserver.go

```diff
@@ -667,6 +683,106 @@ func (d *Driver) DeleteBlobContainer(ctx context.Context, subsID, resourceGroupN
+			if getErr != nil {
+				return true, getErr
+			}
+			srcExist, _ := srcContainer.Exists()
+			if !srcExist {
+				return true, fmt.Errorf("srcContainer(%s) is not exist", srcContainerName)
+			}
+			srcPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, srcContainerName, accountSasToken)
+			dstPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, dstContainerName, accountSasToken)
+
+			klog.V(2).Infof("copy blob container %s to %s", srcContainerName, dstContainerName)
+			out, err := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
+			if err != nil {
+				return true, status.Errorf(codes.Internal, "failed to copy blob container %v: %v", err, string(out))
+			}
+			if waitErr := d.waitForBlobContainerCopy(ctx, subsID, resourceGroupName, accountName, dstContainerName, secrets, waitForBlobContainerCopyInterval, waitForBlobContainerCopyTimeout); waitErr != nil {
+				return true, status.Error(codes.Internal, fmt.Sprintf("failed to wait for blob container copy: %v", waitErr.Error()))
+			}
+			klog.V(2).Infof("copied blob container %s to %s", srcContainerName, dstContainerName)
+		} else {
+			klog.V(2).Infof("generate sas token for account(%s)", accountName)
+			accountSasToken, genErr := GenerateSASToken(accountName, accountKey, blobUrl)
+			if genErr != nil {
+				return true, genErr
+			}
+			_, rerr := d.cloud.BlobClient.GetContainer(ctx, subsID, resourceGroupName, accountName, srcContainerName)
+			if rerr != nil {
+				return true, fmt.Errorf("srcContainer(%s) get error: %s", srcContainerName, rerr.Error())
+			}
+			srcPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, srcContainerName, accountSasToken)
+			dstPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, dstContainerName, accountSasToken)
+
+			klog.V(2).Infof("copy blob container %s to %s", srcContainerName, dstContainerName)
+			out, err := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
+			if err != nil {
+				return true, status.Errorf(codes.Internal, "failed to copy blob container %v: %v", err, string(out))
+			}
+			if waitErr := d.waitForBlobContainerCopy(ctx, subsID, resourceGroupName, accountName, dstContainerName, secrets, waitForBlobContainerCopyInterval, waitForBlobContainerCopyTimeout); waitErr != nil {
+				return true, status.Error(codes.Internal, fmt.Sprintf("failed to wait for blob container copy: %v", waitErr.Error()))
+			}
+			klog.V(2).Infof("copied blob container %s to %s", srcContainerName, dstContainerName)
+		}
+		if err != nil {
+			if strings.Contains(err.Error(), containerBeingDeletedDataplaneAPIError) ||
+				strings.Contains(err.Error(), containerBeingDeletedManagementAPIError) {
+				klog.Warningf("CopyContainer(%s, %s, %s) from(%s, %s, %s) failed with error(%v), retry", resourceGroupName, accountName, dstContainerName, resourceGroupName, accountName, srcContainerName, err)
```

Comment:
what is this for?

Reply:
It's a mistake, if we get error in azcopy command, it will give warning for copy container failed.